src/Microsoft.Sbom.Api/Utils/Comparer/SbomFileComparer.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Sbom.Contracts;
+
+namespace Microsoft.Sbom.Api.Utils.Comparer;
+
+public class SbomFileComparer : IEqualityComparer<SbomFile>
+{
+    private static readonly SbomChecksumComparer ChecksumComparer = new SbomChecksumComparer();
+
+    public bool Equals(SbomFile file1, SbomFile file2)
+    {
+        if (file1 is null || file2 is null)
+        {
+            return false;
+        }
+
+        var licenseInfosEqual = (file1.LicenseInfoInFiles is null && file2.LicenseInfoInFiles is null) ||
+                        file1.LicenseInfoInFiles?.SequenceEqual(file2.LicenseInfoInFiles ?? Enumerable.Empty<string>()) == true;
```

Comment:
Just to be sure, if `file`.LicenseInfoInFiles` is `null`, this ends up returning _false_ because `null != true` ?

Reply:
I understand the "both null is true" case, but the "at least one is not null" case is comparing the raw `file1.LicenseInfoInFiles` against an adjusted `file2.LicenseInfoInFiles`. If the assertion is that a null value and an empty list are equivalent, then you can reduce all cases to the following:

```
var file1LicenseInfoInFiles = file1.LicenseInfoInFiles ?? Enumerable.Empty<string>();
var file2LicenseInfoInFiles = file2.LicenseInfoInFiles ?? Enumerable.Empty<string>();
var licenseInfosEqual = file1LicenseInfoInFiles.SequenceEqual(file2LicenseInfoInFiles);
```